opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -2426,11 +2452,19 @@ namespace Opm
     MultisegmentWell<TypeTag>::
     updateUpwindingSegments()
     {
-        // not considering upwinding for the injectors for now
-        // but we should
-        // and upwinding segment for top segment is itself
         for (int seg = 0; seg < numberOfSegments(); ++seg) {
-            if ( (well_type_ == INJECTOR) || (seg == 0) ) {
+            // special treatment is needed for segment 0
+            if (seg == 0) {
+                if ( (well_type_ == INJECTOR)) {
```

Comment:
The only difference inside `if` and outside the `if` is the `assert`.  

You can use the assertion of combined logic to avoid to have if and duplicating the following code
```
 upwinding_segments_[seg] = seg;
 continue;
```


Something like assert( ! (well_type_ == INJECTOR && primary_variables_evaluation_[seg][GTotal] <= 0.));